src/main/java/com/hazelcast/kubernetes/ServiceEndpointResolver.java

```diff
@@ -142,7 +143,8 @@ private void addAddress(List<DiscoveryNode> discoveredNodes, EndpointAddress end
         Map<String, Object> properties = endpointAddress.getAdditionalProperties();
         String ip = endpointAddress.getIp();
         InetAddress inetAddress = mapAddress(ip);
-        int port = getServicePort(properties);
+        int port = (this.port > 0) ? this.port : getServicePort(properties);
+        logger.fine("Discovered node: " + ip + " port: " + port);
```

Comment:
Please uniform the logging code here and in `DnsEndpointResolver`, i.e., log the port number there and check the log level here.

Reply:
fixed